deploy/install.sh

```diff
@@ -0,0 +1,183 @@
+#cd ..
+
+# Role - Base
+echo "Installing base dependency packages"
+sudo ${package_manager} -y install bc python3-pip curl --quiet
+
+# Role - Docker
+if [[ $package_manager -eq apt-get ]];then
+    echo "++++++++++++++++++++++++"
+    echo "Setting up docker repos"
+    sudo $package_manager update  --quiet
+
+    sudo apt-get  -y --quiet install gnupg-agent
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    sudo add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) \
+    stable"
+else
+    sudo yum install -y yum-utils
+    sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+fi
+
+sudo ${package_manager} -y update --quiet
+echo "Installing docker"
+sudo ${package_manager} -y install docker-ce docker-ce-cli containerd.io --quiet
+
+echo "Installing docker-compose"
+sudo curl -L "https://github.com/docker/compose/releases/download/1.26.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+sudo chmod +x /usr/local/bin/docker-compose
+
+# Role - folders
+ubuntu="/etc/debian_version"
+centos="/etc/centos-release"
+redhat="/etc/redhat-release"
+
+if [ -f $ubuntu ]
+then
+    user="ubuntu"
+    group="ubuntu"
+elif [ -f $centos ]
+then
+   user="centos"
+   group="centos"
+elif [ -f $redhat ]
+then
+   user="redhat"
+   group="redhat"
+fi
+
+for directory_name in nginx certbot mongo/db opa/config appsmith-server/config
+do
+  if [ -d "$install_dir/data/$directory_name" ]
+  then
+    echo "Directory already exists"
+  else
+    mkdir -p "$install_dir/data/$directory_name"
+  fi
+done
+
+echo "Generating the configuration files from the templates"
+. ./template/nginx_app.conf.sh
+. ./template/docker-compose.yml.sh
+. ./template/mongo-init.js.sh
+. ./template/init-letsencrypt.sh.sh
+. ./template/docker.env.sh
+chmod 0755 init-letsencrypt.sh
+
+declare -A fileInfo
+
+fileInfo[/data/nginx/app.conf]="nginx_app.conf"
+fileInfo[/docker-compose.yml]="docker-compose.yml"
+fileInfo[/data/mongo/init.js]="mongo-init.js"
+fileInfo[/init-letsencrypt.sh]="init-letsencrypt.sh"
+fileInfo[/docker.env]="docker.env"
+
+for f in ${!fileInfo[@]}
+do
+
+    if [ -f $install_dir$f ]
+    then
+        echo "File already exist."
+        read -p "File $f already exist. Would you like to replace it? [Y]: " value
+
+        if [ $value == "Y" -o $value == "y" -o $value == "yes" -o $value == "Yes" ]
+        then
+            mv -f  ${fileInfo[$f]} $install_dir$f
+            echo "File $install_dir$f replaced succeffuly!"
+        else
+            echo "You choose not to replae existing file: $install_dir$f"
+	    rm -rf ${fileInfo[$f]}
+	    echo "File ${fileInfo[$f]} removed from source directory."
+        echo ""
+        fi
+    else
+        mv -f ${fileInfo[$f]} $install_dir$f
+    fi
+
+done
+
+
+echo "++++++++++++++++++++"
+
+#echo "Running init-letsencrypt.sh...."
+cd $install_dir
+if [ $custom_domain ];then
```

Comment:
Similar to my other comment in this script file, we probably want the below:

```suggestion
if [[ -z $custom_domain ]]; then
```